### ISSUE TITLE
bluetooth: shell: Lots of checks of type (unsigned < 0) which is bogus

### ISSUE THREAD

**Issue by finikorg** [zephyrproject-rtos/zephyr#54101]
Unsigned cannot be less then zero so there is no need to check for it.

Example of checks:
https://github.com/zephyrproject-rtos/zephyr/blob/cf871aec645dc45503b20c2e7684a4babbc436dc/subsys/bluetooth/shell/bap_broadcast_assistant.c#L315-L316
 or
https://github.com/zephyrproject-rtos/zephyr/blob/cf871aec645dc45503b20c2e7684a4babbc436dc/subsys/bluetooth/shell/bap_broadcast_assistant.c#L321-L322

etc

**Comment by Thalley** [comment]
@finikorg Do you have a way to generate the full list? 

It's quite an effort if someone manually have to go through each `strtoX` in the shell modules to verify every single parameter.

**Comment by finikorg** [comment]
No, I cannot, but I think it does make sense to check them all, there are other bugs like checking (uint32 > UINT32_MAX), etc

**Comment by Thalley** [comment]
Fixing all LE Audio (and ISO) shell `strtoX` in https://github.com/zephyrproject-rtos/zephyr/pull/54165

**Comment by Thalley** [comment]
@finikorg I think I fixed all the bugs in the LE audio files. Please review :)

**Comment by Thalley** [comment]
@finikorg Can you verify if there are still some missing in the non-LE Audio files?

**Comment by finikorg** [comment]
> Can you verify if there are still some missing in the non-LE Audio files?

Thanks. I think all fixed, I let you know if there are some bugs left.
